pkg/proxy/request.go

```diff
@@ -159,3 +180,50 @@ func fileToString(f io.ReadCloser) (string, error) {
 
 	return s, nil
 }
+
+func getTags(r *http.Request) (map[string]string, error) {
+	o := r.FormValue(tags)
+
+	if o == "" {
+		return nil, nil
+	}
+
+	pairs := strings.Split(strings.TrimSpace(o), ",")
+	tags := make(map[string]string, len(pairs))
+
+	for _, pair := range pairs {
+		pair = strings.TrimSpace(pair)
+		if len(pair) < 1 {
+			continue
+		}
+
+		parts := strings.SplitN(pair, ":", 2)
+		label := strings.TrimSpace(parts[0])
+
+		if len(label) == 0 {
+			return nil, ErrTagMissingLabel
+		}
+
+		if len(parts) != 2 {
+			return nil, ErrTagMissingValue
+		}
+
+		value := strings.TrimSpace(parts[1])
+
+		if len(value) == 0 {
+			return nil, ErrTagMissingValue
+		}
+
+		if len(value) > maxTagLength {
+			return nil, ErrTagValueMaxLengthExceeded
+		}
+
+		tags[label] = value
```

Comment:
This looks very similar to https://github.com/hellofresh/kangal/pull/39/files#diff-0530307c704a9dc589d79d6e98bd73aaR85-R108 and https://github.com/hellofresh/kangal/pull/39/files#diff-047152d6343305cf4a2ae0cf8a7611b3R735-R763 
Is there a reason why we can't fine a way to avoid this duplication? Maybe introduce a `type Tags map[string]string` and a constructor method for it?

Reply:
This one is for parsing the input from POST request, we're forming the tags and validate it at the same time